src/lib.rs

```diff
@@ -316,7 +325,16 @@ pub async fn collect_from_pods(
             context: "list pods".to_string(),
             source,
         })?;
-    for pod in pods.items.into_iter().filter(is_scheduled) {
+    compute_pod_utilizations(pods, resources).await?;
+    Ok(())
+}
+
+#[instrument(skip(pod_list, resources))]
+pub async fn compute_pod_utilizations(
+    pod_list: ObjectList<Pod>,
+    resources: &mut Vec<Resource>,
+) -> Result<(), Error> {
```

Comment:
wrong name, no utilization info are computed. only resources allocations (request and limit).
A better name.
A better name could be `search_allocations_from_pos` (nothing related with utilization) (search or extract,... but not compute because too vague and no real mathematics computation), use the same verb that for the similar task.